client/src/features/connectedServices/connectedServices.api.ts

```diff
@@ -0,0 +1,90 @@
+/*!
+ * Copyright 2024 - Swiss Data Science Center (SDSC)
+ * A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+ * Eidgenössische Technische Hochschule Zürich (ETHZ).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+
+import {
+  ProviderList,
+  ConnectionList,
+  ConnectedAccount,
+  GetConnectedAccountParams,
+} from "./connectedServices.types";
+
+const connectedServicesApi = createApi({
+  reducerPath: "connectedServicesApi",
+  baseQuery: fetchBaseQuery({
+    baseUrl: "/ui-server/api/data",
```

Comment:
Nitpick: all routes go through `oauth2`. We could move it up here to make it more obvious

```suggestion
    baseUrl: "/ui-server/api/data/oauth2",
```